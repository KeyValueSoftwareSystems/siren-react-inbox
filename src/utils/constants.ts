export const COLORS = {
  light: {
    primaryColor: "#F56630",
    highlightedCardColor: "#FFECE5",
    textColor: "#344054",
    neutralColor: "#FFFFFF",
    borderColor: "#D0D5DD",
    dateColor: "#34405499",
    deleteIcon: "#34405499",
    timerIcon: "#667185",
    clearAllIcon: "#667185",
    windowShadowColor: "#D0D5DD",
    iconColor: "#F7F9FC",
    notificationIcon: "#232326",
  },
  dark: {
    primaryColor: "#F56630",
    highlightedCardColor: "#2E2D30",
    textColor: "#FFFFFF",
    neutralColor: "#232326",
    borderColor: "#344054",
    dateColor: "#98A2B3",
    deleteIcon: "#D0D5DD",
    timerIcon: "#98A2B3",
    clearAllIcon: "#D0D5DD",
    windowShadowColor: "#232326",
    iconColor: "#38383D",
    notificationIcon: "#FFFFFF",
  },
};

export enum ThemeMode {
  DARK = "dark",
  LIGHT = "light",
}

export enum LogLevel {
  INFO,
  ERROR,
}

export enum BadgeType {
  NONE = "none",
  DOT = "dot",
  DEFAULT = "default",
}

export const levelLogFns = {
  [LogLevel.INFO]: console.log,
  [LogLevel.ERROR]: console.error,
};

export const defaultBadgeStyle = {
  size: 15,
  color: "#FF0000",
  textColor: "#FFFFFF",
  textSize: 10,
  linHeight: "14px",
  top: "inherit",
  left: "inherit",
};

export enum eventTypes {
  MARK_ITEM_AS_VIEWED = "MARK_ITEM_AS_VIEWED",
  MARK_ITEM_AS_READ = "MARK_ITEM_AS_READ",
  MARK_ALL_AS_READ = "MARK_ALL_AS_READ",
  DELETE_ITEM = "DELETE_ITEM",
  DELETE_ALL_ITEM = "DELETE_ALL_ITEM",
  NEW_NOTIFICATIONS = "NEW_NOTIFICATIONS",
  UPDATE_NOTIFICATIONS_COUNT = "UPDATE_NOTIFICATIONS_COUNT",
  RESET_NOTIFICATIONS = "RESET_NOTIFICATIONS",
  RESET_NOTIFICATIONS_COUNT = "RESET_NOTIFICATIONS_COUNT",
}

export enum events {
  NOTIFICATION_LIST_EVENT = "NOTIFICATION_LIST_EVENT",
  NOTIFICATION_COUNT_EVENT = "NOTIFICATION_COUNT_EVENT",
}

export enum updateNotificationsTypes {
  MARK_ITEM_AS_VIEWED = "MARK_ITEM_AS_VIEWED",
  MARK_ITEM_AS_READ = "MARK_ITEM_AS_READ",
  MARK_ALL_AS_READ = "MARK_ALL_AS_READ",
  DELETE_ITEM = "DELETE_ITEM",
  DELETE_ALL_ITEM = "DELETE_ALL_ITEM",
}

export enum sirenReducerTypes {
  SET_NOTIFICATIONS = "SET_NOTIFICATIONS",
  NEW_NOTIFICATIONS = "NEW_NOTIFICATIONS",
  SET_UN_VIEWED_NOTIFICATION_COUNT = "SET_UN_VIEWED_NOTIFICATION_COUNT",
  SET_SIREN_CORE = "SET_SIREN_CORE",
}

export const LIST_EMPTY_TEXT = "No new notifications";
export const LIST_EMPTY_SUB_TEXT = "Check back later for updates and alerts.";
export const ERROR_TEXT = "Oops! Something went wrong";
export const ERROR_SUB_TEXT =
  "Could not load the notifications. Please refresh the page.";
export const DEFAULT_WINDOW_TITLE = "Notifications";
export const RETRY_BUTTON_LABEL = "Retry";
export const CLEAR_ALL_LABEL = "Clear All";
export const IN_APP_RECIPIENT_UNAUTHENTICATED = 'IN_APP_RECIPIENT_UNAUTHENTICATED';
export const TOKEN_VERIFICATION_PENDING ='TOKEN_VERIFICATION_PENDING';
export const MAXIMUM_RETRY_COUNT = 3;

export const errorMap = {
  SIREN_OBJECT_NOT_FOUND: {
    Type: "ERROR",
    Code: "SIREN_OBJECT_NOT_FOUND",
    Message: "Siren Object Not found",
  },
  MISSING_PARAMETER: {
    Type: "ERROR",
    Code: "MISSING_PARAMETER",
    Message: "Missing Parameter",
  },
};

export enum VerificationStatus {
  PENDING = "PENDING",
  SUCCESS = "SUCCESS",
  FAILED = "FAILED"
}