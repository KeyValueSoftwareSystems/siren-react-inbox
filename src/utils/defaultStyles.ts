
const defaultStyles = {
  notificationIcon: {
    size: 30,
  },
  window: {
    width: 500,
    borderRadius: "10px",
    borderWidth: "0px",
    borderStyle: "solid"
  },
  windowHeader: {
    height: 64,
    titleFontWeight: "600",
    titleSize: 20,
    closeIconSize: 20,
    titlePadding: 0,
  },
  windowContainer: {
    padding: 0,
    height: 700,
  },
  notificationCard: {
    padding: 10,
    borderWidth: 0.5,
    avatarSize: 40,
    titleFontWeight: "600",
    titleSize: 16,
    descriptionSize: 14,
    dateSize: 12,
  },
  loadMoreButton: {
    fontSize: 14,
    fontWeight: '600'
  },
  badgeStyle: {
    size: 15,
    textSize: 10,
  },
  deleteIcon:{
    size: 16
  },
  timerIcon:{
    size: 12
  },
  clearAllIcon:{
    size: 18
  }
}

export default defaultStyles;
