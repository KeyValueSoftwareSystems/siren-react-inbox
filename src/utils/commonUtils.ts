
import { DefaultTheme } from ".";
import type { SirenStyleProps, ThemeProps } from "../types";
import { LogLevel} from "./constants";
import { ThemeMode } from "./constants";

export const getTimeAgo = (timeString: string) => {
  const currentTime = Date.now();
  const targetTime = new Date(timeString).getTime();
  const timeDiff = currentTime - targetTime;

  const seconds = Math.floor(timeDiff / 1000);
  const minutes = Math.floor(seconds / 60);
  const hours = Math.floor(minutes / 60);
  const days = Math.floor(hours / 24);
  const years = Math.floor(days / 365);

  if (timeDiff < 60000) 
    return 'Just now';
  else if (minutes < 60) 
    return minutes === 1 ? '1 minute ago' : `${minutes} minutes ago`;
  else if (hours < 24) 
    return hours === 1 ? '1 hour ago' : `${hours} hours ago`;
  else if (days < 365) 
    return days === 1 ? '1 day ago' : `${days} days ago`;
  else 
    return years === 1 ? '1 year ago' : `${years} years ago`;
};



export const renderAsImage = (icon: JSX.Element | string): boolean =>
  typeof icon === 'undefined' || typeof icon === 'string';

export const logger = {
  log: async (level:LogLevel.INFO | LogLevel.ERROR, message: string) => {
    const timestamp = new Date().toISOString();
    const levelString = LogLevel[level].toUpperCase();

    console.log(`[${timestamp}] [${levelString}] ${message}`);
  },
  error: function (error: string) {
    this.log(LogLevel.ERROR, error);
  },
  info: function (message: string) {
    this.log(LogLevel.INFO, message);
  }
};
   


export const applyTheme = (
  theme: ThemeProps = {},
  mode: ThemeMode = ThemeMode.DARK,
): SirenStyleProps =>
  ({
    container: {
      width: theme.window?.width || DefaultTheme[mode].window.width,
      minHeight: theme.window?.height || DefaultTheme[mode].window.height,
      borderColor:
          theme.window?.borderColor || DefaultTheme[mode].window.borderColor,
      borderRadius:
          theme.window?.borderRadius || DefaultTheme[mode].window.borderRadius,
      maxWidth: theme.window?.width,
      flex: 1,
      overflow: 'hidden'
    },
    contentContainer: {
      backgroundColor:
          theme.windowContainer?.background ||
          theme.colors?.neutralColor ||
          DefaultTheme[mode].windowContainer.background,
      padding:
          theme.windowContainer?.padding ||
          DefaultTheme[mode].windowContainer.padding,
      overflow: 'auto',
      maxHeight: '700px',
    },
    headerContainer: {
      backgroundColor:
          theme.windowHeader?.background ||
          theme.colors?.primaryColor ||
          DefaultTheme[mode].windowHeader.background,

      height:
          theme.windowHeader?.height || DefaultTheme[mode].windowHeader.height,
      justifyContent: 'space-between',
      padding:'10px 15px 10px 15px',
      borderBottom:' 1px solid #dbdae2',
      display: 'flex',
      flexDirection: 'row'
    },
    headerTitle: {
      alignItems: 'center',
      display: 'flex',
      margin: 0,
      color:
          theme.windowHeader?.titleColor ||
          theme.colors?.neutralColor ||
          DefaultTheme[mode].windowHeader.titleColor,
      fontSize:
          theme.windowHeader?.titleSize ||
          DefaultTheme[mode].windowHeader.titleSize,
      fontWeight:
          theme.windowHeader?.titleFontWeight ||
          DefaultTheme[mode].windowHeader.titleFontWeight,
    },
    headerButton:{
      outline: 'none',
      border: 'none',
      fontWeight: 700,
      cursor: 'pointer',
      alignItems: 'center',
      display: 'flex'
    },
    headerAction: {
      color:
        theme.windowHeader?.titleColor ||
        theme.colors?.neutralColor ||
        DefaultTheme[mode].windowHeader.titleColor,
    },
    cardContainerButton: {
      backgroundColor:
          theme.notificationCard?.background ||
          theme.colors?.neutralColor ||
          DefaultTheme[mode].notificationCard.background,
      padding:
          theme.notificationCard?.padding ||
          DefaultTheme[mode].notificationCard.padding,
      flexDirection: 'row',
      borderBottom: `${theme.notificationCard?.borderWidth || DefaultTheme[mode].notificationCard.borderWidth}px solid`,
      borderColor:
      theme.notificationCard?.borderColor ||
      theme.colors?.borderColor ||
      DefaultTheme[mode].notificationCard.borderColor,
    },
    cardContainer: {
      display: 'flex',
      flexDirection: 'column',
      width: '100%'
    },
    cardIconRound: {
      width:
          theme.notificationCard?.avatarSize ||
          DefaultTheme[mode].notificationCard.avatarSize,
      height:
          theme.notificationCard?.avatarSize ||
          DefaultTheme[mode].notificationCard.avatarSize,
      borderRadius:
          (theme.notificationCard?.avatarSize ||
            DefaultTheme[mode].notificationCard.avatarSize) / 2,
      overflow: 'hidden',
      backgroundColor:
          theme.colors?.borderColor || DefaultTheme[mode].colors.borderColor,
    },
    cardTitle: {
      margin: 0,
      color:
          theme.notificationCard?.titleColor ||
          theme.colors?.textColor ||
          DefaultTheme[mode].notificationCard.titleColor,
      fontSize:
          theme.notificationCard?.titleSize ||
          DefaultTheme[mode].notificationCard.titleSize,
      fontWeight:
          theme.notificationCard?.titleFontWeight ||
          DefaultTheme[mode].notificationCard.titleFontWeight,
      paddingBottom: 4,
      paddingRight:
          theme.notificationCard?.padding ||
          DefaultTheme[mode].notificationCard.titlePadding,
      paddingLeft:  theme.notificationCard?.padding ||
      DefaultTheme[mode].notificationCard.titlePadding,
    },
    cardContentWrapper: {
      display: 'flex',
      justifyContent: 'space-between',
      width: '100%',
    },
    cardDescription: {
      margin: 0,
      color:
          theme.notificationCard?.descriptionColor ||
          theme.colors?.textColor ||
          DefaultTheme[mode].notificationCard.descriptionColor,
      fontSize:
          theme.notificationCard?.descriptionSize ||
          DefaultTheme[mode].notificationCard.descriptionSize,
      fontWeight: '400',
      paddingBottom: 10,
      paddingRight:
          theme.notificationCard?.descriptionPadding ||
          DefaultTheme[mode].notificationCard.descriptionPadding,
      paddingLeft: theme.notificationCard?.descriptionPadding ||
      DefaultTheme[mode].notificationCard.descriptionPadding,
    },
    cardImageStyle: {
      objectFit: 'cover',
      width:
          theme.notificationCard?.mediaWidth ||
          DefaultTheme[mode].notificationCard.mediaWidth,
      height:
          theme.notificationCard?.mediaHeight ||
          DefaultTheme[mode].notificationCard.mediaHeight,
      borderRadius:
          theme.notificationCard?.mediaRadius ||
          DefaultTheme[mode].notificationCard.mediaRadius,
      overflow: 'hidden',
      marginBottom: 10,
      backgroundColor:
          theme.notificationCard?.mediaPlaceholder ||
          DefaultTheme[mode].notificationCard.mediaPlaceholder,
    },
    dateStyle: {
      color:
          theme.notificationCard?.dateColor ||
          theme.colors?.borderColor ||
          DefaultTheme[mode].notificationCard.dateColor,
      fontSize:
          theme.notificationCard?.dateSize ||
          DefaultTheme[mode].notificationCard.dateSize,
      opacity: 0.6,
      margin: 0,
    },
    deleteButton: {
      width: '20px',
      height: '20px',
      padding: '4px',
      cursor: 'pointer'
    },
    footerContainer: {
      backgroundColor:
      theme.windowFooter?.background ||
      theme.colors?.primaryColor ||
      DefaultTheme[mode].windowFooter.background,

      height:
      theme.windowFooter?.height || DefaultTheme[mode].windowFooter.height,
      color:
          theme.windowFooter?.textColor ||
          theme.colors?.neutralColor ||
          DefaultTheme[mode].windowFooter.textColor,
      fontSize:
          theme.windowFooter?.textSize ||
          DefaultTheme[mode].windowFooter.textSize,
      fontWeight:
          theme.windowFooter?.textFontWeight ||
          DefaultTheme[mode].windowFooter.textFontWeight,
      borderColor:
          theme.window?.borderColor || DefaultTheme[mode].window.borderColor,
      width: '100%',
      position: 'relative',
      cursor: 'pointer',
      border: 'none',
      justifyContent: 'center',
      alignItems: 'center',
      display: 'flex',
    },
    notificationBadgeContainer:{
      backgroundColor: theme?.unreadBadgeCount?.background || 
                          DefaultTheme[mode].unreadBadgeCount.background,
      borderRadius: theme?.unreadBadgeCount?.borderRadius ||
                       DefaultTheme[mode].unreadBadgeCount.borderRadius , 
      height: theme?.unreadBadgeCount?.size ||
      DefaultTheme[mode].unreadBadgeCount.size,
      minWidth:  theme?.unreadBadgeCount?.size ||
      DefaultTheme[mode].unreadBadgeCount.size,
      position: 'absolute',
      top: 0,
      right: '6px',
      display: 'flex',
      justifyContent: 'center'
    },
    notificationCount: {
      color: theme?.unreadBadgeCount?.color || 
      DefaultTheme[mode].unreadBadgeCount.color,
      fontSize: theme?.unreadBadgeCount?.fontSize || 
      DefaultTheme[mode].unreadBadgeCount.fontSize,
    },
    emptyContainer: {
      justifyContent: 'center',
      alignItems: 'center',
      flex: 1,
      minHeight: 100,
      width: '100%',
    },
    emptyText: {
      color: theme.colors?.textColor || DefaultTheme[mode].colors.textColor,
      fontSize: 18,
      fontWeight: '400',
      padding: 20,
    },
    errorText: {
      color: theme.colors?.primaryColor || DefaultTheme[mode].colors.primaryColor,
      fontSize: 18,
      fontWeight: '400',
      padding: 20,
    },
    errorButton:{
      padding: '6px 20px 6px 20px',
      justifyContent: 'center',
      alignItems: 'center',
      borderRadius: 5,
      overflow: 'hidden',
      backgroundColor: theme.colors?.primaryColor || DefaultTheme[mode].colors.primaryColor,
    },
    errorButtonText:{
      color: theme.colors?.primaryTextColor || DefaultTheme[mode].colors.primaryTextColor,
      fontSize: 16,
      fontWeight: '500',
    },
    sirenWindowLoaderContainer :{
      border: '8px solid #f3f3f3',
      borderTop: '8px solid #3498db',
      borderRadius: '50%',
      width: '10px',
      height: '10px',
      animation: 'spin 1s linear infinite',
      margin: 0 ,
    },
    // sirenWindowLoader:{
    //   width: '100%',
    //   position: 'relative ',
    //   cursor: 'pointer',
    //   border: 'none'
    // }
    avatarContainer: {
      display: 'flex',
      width:  '100%'
    },
    avatarImageContainer: {
      display: 'flex',
      justifyContent: 'center',
      paddingRight: '16px'
    },
    cardMediaContainer: {
      display: 'flex',
      justifyContent: 'center'
    },
    notificationIconContainer: {
      border: 'none',
      backgroundColor: 'initial',
      position: 'relative'
    },
    notificationIcon:{
      width: theme?.notificationIcon?.size || DefaultTheme[mode]?.notificationIcon?.size,
      height: theme?.notificationIcon?.size || DefaultTheme[mode]?.notificationIcon?.size,
      padding: '4px'
    }
  });