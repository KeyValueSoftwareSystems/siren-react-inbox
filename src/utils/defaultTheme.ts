import { COLORS, ThemeMode } from "./constants";


const defaultTheme = {
  light: {
    notificationIcon: {
      size: 30,
    },
    colors: {
      primaryColor: COLORS[ThemeMode.LIGHT].primaryColor,
      secondaryColor: COLORS[ThemeMode.LIGHT].secondaryColor,
      textColor: COLORS[ThemeMode.LIGHT].textColor,
      neutralColor: COLORS[ThemeMode.LIGHT].neutralColor,
      borderColor: COLORS[ThemeMode.LIGHT].borderColor,
      primaryTextColor: COLORS[ThemeMode.DARK].primaryTextColor
    },
    unreadBadgeCount: {
      background: COLORS[ThemeMode.LIGHT].secondaryColor,
      color: COLORS[ThemeMode.LIGHT].neutralColor,
      borderRadius: '50%',
      fontSize: 14,
      inset: 2,
      size: 16,
    },
    window: {
      width: '400px',
      height: '100%',
      borderColor: COLORS[ThemeMode.LIGHT].borderColor,
      borderRadius: '5px',
      shadowDepth: 0,
      shadowColor: COLORS[ThemeMode.LIGHT].borderColor,
    },
    windowHeader: {
      background: COLORS[ThemeMode.LIGHT].primaryColor,
      height: 50,
      titleColor: COLORS[ThemeMode.LIGHT].neutralColor,
      titleFontWeight: '500',
      titleSize: 20,
      closeIconColor: COLORS[ThemeMode.LIGHT].neutralColor,
      closeIconSize: 20,
    },
    windowContainer: {
      background: COLORS[ThemeMode.LIGHT].neutralColor,
      padding: 0,
    },
    notificationCard: {
      height: '100%',
      padding: 10,
      borderWidth: 1,
      borderColor: COLORS[ThemeMode.LIGHT].borderColor,
      background: COLORS[ThemeMode.LIGHT].activeCardColor,
      hoverBackground: COLORS[ThemeMode.LIGHT].neutralColor,
      avatarSize: 40,
      titleColor: COLORS[ThemeMode.LIGHT].textColor,
      titleFontWeight: '500',
      titleSize: 16,
      titlePadding: 0,
      descriptionColor: COLORS[ThemeMode.LIGHT].textColor,
      descriptionSize: 14,
      descriptionPadding: 0,
      mediaWidth: '100%',
      mediaHeight: 150,
      mediaObjectFit: 'cover',
      mediaRadius: 8,
      mediaPlaceholder: COLORS[ThemeMode.LIGHT].borderColor,
      dateColor: COLORS[ThemeMode.LIGHT].textColor,
      dateSize: 12,
    },
    windowFooter:{
      background: COLORS[ThemeMode.LIGHT].primaryColor,
      height: 50,
      textColor: COLORS[ThemeMode.LIGHT].primaryTextColor,
      textFontWeight: '500',
      textSize: 14,
    }
  },
  dark: {
    notificationIcon: {
      size: 30,
    },
    colors: {
      primaryColor: COLORS[ThemeMode.DARK].primaryColor,
      secondaryColor: COLORS[ThemeMode.DARK].secondaryColor,
      textColor: COLORS[ThemeMode.DARK].textColor,
      neutralColor: COLORS[ThemeMode.DARK].neutralColor,
      borderColor: COLORS[ThemeMode.DARK].borderColor,
      primaryTextColor: COLORS[ThemeMode.DARK].primaryTextColor
    },
    unreadBadgeCount: {
      background: COLORS[ThemeMode.DARK].secondaryColor,
      color: COLORS[ThemeMode.DARK].neutralColor,
      borderRadius: 10,
      fontSize: 12,
      inset: 2,
      size: 14,
    },
    window: {
      width: '400px',
      height: '100%',
      borderColor: COLORS[ThemeMode.DARK].borderColor,
      borderRadius: '5px',
      shadowDepth: 0,
      shadowColor: COLORS[ThemeMode.DARK].borderColor,
    },
    windowHeader: {
      background: COLORS[ThemeMode.DARK].primaryColor,
      height: 50,
      titleColor: COLORS[ThemeMode.DARK].primaryTextColor,
      titleFontWeight: '500',
      titleSize: 20,
      closeIconColor: COLORS[ThemeMode.DARK].neutralColor,
      closeIconSize: 20,
    },
    windowContainer: {
      background: COLORS[ThemeMode.DARK].neutralColor,
      padding: 0,
    },
    notificationCard: {
      height: '100%',
      padding: 10,
      borderWidth: 1,
      borderColor: COLORS[ThemeMode.DARK].borderColor,
      background: COLORS[ThemeMode.DARK].activeCardColor,
      hoverBackground: COLORS[ThemeMode.DARK].neutralColor,
      avatarSize: 40,
      titleColor: COLORS[ThemeMode.DARK].textColor,
      titleFontWeight: '500',
      titleSize: 16,
      titlePadding: 0,
      descriptionColor: COLORS[ThemeMode.DARK].textColor,
      descriptionSize: 14,
      descriptionPadding: 0,
      mediaWidth: '100%',
      mediaHeight: 150,
      mediaObjectFit: 'cover',
      mediaRadius: 8,
      mediaPlaceholder: COLORS[ThemeMode.DARK].borderColor,
      dateColor: COLORS[ThemeMode.DARK].textColor,
      dateSize: 12,
    },
    windowFooter:{
      background: COLORS[ThemeMode.DARK].primaryColor,
      height: 50,
      textColor: COLORS[ThemeMode.DARK].primaryTextColor,
      textFontWeight: '500',
      textSize: 14,
    }
  },
};

export default defaultTheme;