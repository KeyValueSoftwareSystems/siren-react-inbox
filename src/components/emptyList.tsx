import React from 'react';

import { LIST_EMPTY_TEXT } from '../utils/constants';
import type { SirenStyleProps } from '../types';

/**
 * EmptyList component represents a placeholder displayed when the notification list is empty.
 *
 * @component
 * @example
 * <EmptyList
 *   styles={customStyles}
 * />
 *
 * @param {Object} props - The properties passed to the EmptyList component.
 * @param {Object} props.styles - The styles object to customize the appearance of the empty list.
 * @returns {ReactElement} The rendered EmptyList component.
 */
const EmptyList = (props: { styles: SirenStyleProps }) => {
  const { styles } = props;

  return (
    <div style={styles.emptyContainer}>
      <p style={styles.emptyText}>{LIST_EMPTY_TEXT}</p>
    </div>
  );
};

export default EmptyList;