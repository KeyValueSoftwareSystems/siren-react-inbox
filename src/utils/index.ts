import * as CommonUtils from './commonUtils';
import * as Constants from './constants';
import DefaultStyle from './defaultStyles'
import DefaultTheme from './defaultTheme';
import useSiren from './sirenHook';

export {
  Constants,
  DefaultTheme,
  useSiren,
  CommonUtils,
  DefaultStyle
};