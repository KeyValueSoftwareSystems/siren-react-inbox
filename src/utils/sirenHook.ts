import PubSub from "pubsub-js";

import { errorMap, events, eventTypes } from "./constants";
import { useSirenContext } from "../components/SirenProvider";

const useSiren = () => {
  const { siren, id: providerInstanceId } = useSirenContext();

  const markAsReadById = async (id: string) => {
    if (siren)
      if (id?.length > 0) {
        const response = await siren?.markAsReadById(id);

        if (response?.data) {
          const payload = { id, action: eventTypes.MARK_ITEM_AS_READ };

          PubSub.publish(
            `${events.NOTIFICATION_LIST_EVENT}${providerInstanceId}`,
            JSON.stringify(payload)
          );
        }

        return response;
      } else {
        return { error: errorMap.MISSING_PARAMETER };
      }

    return { error: errorMap.SIREN_OBJECT_NOT_FOUND };
  };

  const markAsReadByDate = async (untilDate: string, category?: string) => {
    if (siren && untilDate) {
      const params = { startDate: untilDate, category };
      const response = await siren?.markAsReadByDate(params);

      if (response?.data) {
        const payload = { action: eventTypes.MARK_ALL_AS_READ };

        PubSub.publish(`${events.NOTIFICATION_LIST_EVENT}${providerInstanceId}`, JSON.stringify(payload));
      }

      return response;
    }

    return { error: errorMap.SIREN_OBJECT_NOT_FOUND };
  };
  
  const deleteById = async (id: string, shouldUpdateList: boolean = true) => {
    if (siren)
      if (id?.length > 0) {
        const response = await siren?.deleteById(id);

        if (response?.data && shouldUpdateList) {
          const payload = { id, action: eventTypes.DELETE_ITEM };

          PubSub.publish(
            `${events.NOTIFICATION_LIST_EVENT}${providerInstanceId}`,
            JSON.stringify(payload)
          );
        }

        return response;
      } else {
        return { error: errorMap.MISSING_PARAMETER };
      }

    return { error: errorMap.SIREN_OBJECT_NOT_FOUND };
  };

  const deleteByDate = async (untilDate: string, category?: string) => {
    if (siren && untilDate) {
      const params = { startDate: untilDate, category };
      const response = await siren?.deleteByDate(params);

      if (response?.data) {
        const payload = { action: eventTypes.DELETE_ALL_ITEM };

        PubSub.publish(`${events.NOTIFICATION_LIST_EVENT}${providerInstanceId}`, JSON.stringify(payload));
      }

      return response;
    }

    return { error: errorMap.SIREN_OBJECT_NOT_FOUND };
  };

  const markAllAsViewed = async (untilDate: string) => {
    if (siren && untilDate) {
      const response = await siren?.markAllAsViewed(untilDate);

      if (response && response.data) {
        const payload = {
          notificationsCount: 0,
          action: eventTypes.UPDATE_NOTIFICATIONS_COUNT,
        };

        PubSub.publish(
          `${events.NOTIFICATION_COUNT_EVENT}${providerInstanceId}`,
          JSON.stringify(payload)
        );
      }

      return response;
    }

    return { error: errorMap.SIREN_OBJECT_NOT_FOUND };
  };

  return {
    markAsReadByDate,
    markAsReadById,
    deleteById,
    deleteByDate,
    markAllAsViewed,
  };
};

export default useSiren;
