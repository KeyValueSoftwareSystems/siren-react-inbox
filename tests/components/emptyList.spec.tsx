import React from "react";
import { render } from "@testing-library/react";
import EmptyList from "../../src/components/emptyList";
import { applyTheme } from "../../src/utils/commonUtils";

const style = applyTheme();

test("matches snapshot", () => {
  const { asFragment } = render(<EmptyList styles={style} />);

  expect(asFragment()).toMatchSnapshot();
});