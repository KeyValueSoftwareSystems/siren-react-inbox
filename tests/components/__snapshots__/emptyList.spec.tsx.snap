// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    style="justify-content: center; align-items: center; flex: 1; min-height: 100px; width: 100%;"
  >
    <p
      style="color: rgb(255, 255, 255); font-size: 18px; font-weight: 400; padding: 20px;"
    >
      You don't have any notifications!
    </p>
  </div>
</DocumentFragment>
`;
